### PR TITLE
Support v-if multiple conditions

### PR DESCRIPTION
**Pull request by defcc**
Support v-if multiple conditions

```html
<div v-if="tester1"></div>
<div v-elseif="tester2"></div>
<div v-elseif="tester3"></div>
<div v-else></div>
```

Collects `if` and `v-elseif` and `v-else` into element.conditions when parsing and generates code like (a)?_h:(c)?_h:_e()

### CODE REVIEW

**Comment by blake-newman** [comment]
`v-elseif` or `v-else-if`?

Latter might be better for non-native users

File: flow/compiler.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ declare type ModuleOptions = {
 }
 
 declare type ASTModifiers = { [key: string]: boolean }
+declare type ASTIfConditions = Array<{ exp: ?string; block: ASTElement }>
```
Comment (zzz6519003) [inline flow/compiler.js]: what's this doing

File: test/unit/features/directives/if.spec.js [hunk 2]
```diff
@@ -88,6 +88,55 @@ describe('Directive v-if', () => {
...
+          <span v-if="foo">hello</span>
+          <span v-elseif="bar">elseif</span>
+          <span v-else>bye</span>
+        </div>
+      `,
+      data: { foo: true, bar: false }
+    }).$mount()
+    expect(vm.$el.innerHTML.trim()).toBe('<span>hello</span>')
+    vm.foo = false
+    waitForUpdate(() => {
```
Comment (zzz6519003) [inline test/unit/features/directives/if.spec.js]: what's the waitForUpdate magic

File: test/unit/features/directives/if.spec.js [hunk 3]
```diff
@@ -88,6 +88,55 @@ describe('Directive v-if', () => {
...
+      expect(vm.$el.innerHTML.trim()).toBe('<span>bye</span>')
+      vm.bar = true
+    }).then(() => {
+      expect(vm.$el.innerHTML.trim()).toBe('<span>elseif</span>')
+      vm.bar = false
```
Comment (zzz6519003) [inline test/unit/features/directives/if.spec.js]: if it wants a then chaining,
```js
+      expect(vm.$el.innerHTML.trim()).toBe('<span>elseif</span>')
+      vm.bar = false
```
should return a promise,
but I see no promise T.T

Comment (zzz6519003) [inline test/unit/features/directives/if.spec.js]: @fnlctrl help wanted XD

File: test/unit/features/directives/if.spec.js [hunk 4]
```diff
@@ -88,6 +88,55 @@ describe('Directive v-if', () => {
...
+    }).then(() => {
+      expect(vm.$el.innerHTML.trim()).toBe('<span>hello</span>')
+      vm.foo = false
+      vm.bar = {}
+    }).then(() => {
+      expect(vm.$el.innerHTML.trim()).toBe('<span>elseif</span>')
+      vm.bar = 0
+    }).then(() => {
+      expect(vm.$el.innerHTML.trim()).toBe('<span>bye</span>')
+      vm.bar = []
```
Comment (zzz6519003) [inline test/unit/features/directives/if.spec.js]: why would it be necessary for so many kind of `vm.bar` @fnlctrl

**Pull request merged**